corehq/apps/app_manager/static/app_manager/js/multimedia_size_util.js

```diff
@@ -0,0 +1,44 @@
+hqDefine('app_manager/js/multimedia_size_util',[
+    'jquery',
+    'underscore',
+    'knockout',
+    'hqwebapp/js/initial_page_data',
+], function ($, _, ko, initialPageData) {
+    var multimediaSize = function (name, size) {
+        var self = {};
+        self.name = ko.observable(name);
+        self.size = ko.observable(size);
+        return self;
+    }
+    var multimediaSizesView = function (url) {
+        var self = {};
+        self.sizes = ko.observableArray();
+        self.load_state = ko.observable(null);
+        self.showSpinner = ko.observable(false);
+        self.load = function () {
+            self.load_state('loading');
+            $.ajax({
+                url: url,
+                success: function (content) {
+                    _.each(content, function(mmsize, mmType) {
+                        self.sizes.push(multimediaSize(mmType, mmsize));
```

Comment:
Any reason you're pushing one element at a time instead of replacing all of `self.sizes` in one go?

Reply:
I need to convert to `multimediaSize` object so iteration would be needed right?

Reply:
Yeah, but you could do one statement like
```
self.sizes(_.map(_.keys(content), function (mmType) {
    return multimediaSize(mmType, content[mmType]);
}));
```

Food for thought: the edition of Refactoring that the book club is reading calls loops a code smell and argues they should be replaced with pipelines like `map`. I don't think that that **all** loops are bad but do find myself usually favoring mapping, reducing, etc.

Reply:
Not sure how it better but looks more modern for sure, so going for it
https://github.com/dimagi/commcare-hq/pull/25864/commits/24efb4be61cde2ba8983f1ad3eaa88be92a52af1